src/main/groovy/com/twcable/grabbit/server/batch/steps/jcrnodes/JcrNodesProcessor.groovy

```diff
@@ -194,168 +185,25 @@ class JcrNodesProcessor implements ItemProcessor<JcrNode, Node> {
-                    Value value = jcrProperty.value
-                    propertyBuilder.setValue(NodeProtos.Value.newBuilder().setStringValue(value.string))
-                }
-                else {
-                    Value[] values = jcrProperty.values
-                    Collection<NodeProtos.Value> protoValues = values.collect { Value value ->
-                        NodeProtos.Value.newBuilder().setStringValue(value.string).build()
-                    }
-                    propertyBuilder.setValues(
-                        NodeProtos.Values.newBuilder().addAllValue(protoValues).build()
-                    )
-                }
-                break
-            case PATH:
-                if (!jcrProperty.multiple) {
-                    Value value = jcrProperty.value
-                    propertyBuilder.setValue(NodeProtos.Value.newBuilder().setStringValue(value.string))
-                }
-                else {
-                    Value[] values = jcrProperty.values
-                    Collection<NodeProtos.Value> protoValues = values.collect { Value value ->
-                        NodeProtos.Value.newBuilder().setStringValue(value.string).build()
-                    }
-                    propertyBuilder.setValues(
-                        NodeProtos.Values.newBuilder().addAllValue(protoValues).build()
-                    )
-                }
-                break
-            case PropertyType.URI:
-                if (!jcrProperty.multiple) {
-                    Value value = jcrProperty.value
-                    propertyBuilder.setValue(NodeProtos.Value.newBuilder().setStringValue(value.string))
-                }
-                else {
-                    Value[] values = jcrProperty.values
-                    Collection<NodeProtos.Value> protoValues = values.collect { Value value ->
-                        NodeProtos.Value.newBuilder().setStringValue(value.string).build()
-                    }
-                    propertyBuilder.setValues(
-                        NodeProtos.Values.newBuilder().addAllValue(protoValues).build()
-                    )
-                }
-                break
-        //TODO: Is it correct to ignore this? (as Reference value would mean different for Server and for Client
-            case REFERENCE:
-                break
-        //TODO: Is it correct to ignore this? (seems similar to REFERENCE to me)
-            case WEAKREFERENCE:
-                break
+        if(jcrProperty.type == BINARY) {
+            propertyBuilder.addValues(valueBuilder.setBytesValue(ByteString.copyFrom(jcrProperty.value.binary.stream.bytes)))
+        }
+        else {
+            //Other property types can potentially have multiple values
+            final Value[] values = jcrProperty.multiple ? jcrProperty.values : [jcrProperty.value] as Value[]
+            values.each { Value value ->
+                propertyBuilder.addValues(valueBuilder.setStringValue(value.string))
```

Comment:
Yeah, it's calling getString for all types except binary values.

If you look under the hood, and using your Date example; Jackrabbit implements org.apache.jackrabbit.value.DateValue, which extends org.apache.jackrabbit.value.BaseValue, which implements the Value spec (javax.jcr.Value).  DateValue is just a nice wrapper around BaseValue, and internally does essentially the same thing we were doing before. 

BaseValue.getString() calls an internal abstract method getInternalString(), and DateValue simply implements this by calling ISO8601.format() on it's internal Date reference.

This way, you don't really need to know the subtype of Value in order to interact with getString(); you can expect consistent behavior between a DateValue, and StringValue when calling getString. Even BinaryValue.getString() creates a byte array, and interprets it as a string using the default encoding.